blocksuite/affine/blocks/block-root/src/edgeless/configs/toolbar/render-linked-doc.ts

```diff
@@ -33,27 +33,23 @@ export function createLinkedDocFromNote(
   docTitle?: string
 ) {
   const _doc = doc.workspace.createDoc();
-  const linkedDoc = _doc.getStore();
+  const transformer = doc.getTransformer([
+    replaceIdMiddleware(doc.workspace.idGenerator),
+  ]);
+  const blockSnapshot = transformer.blockToSnapshot(note);
+  if (!blockSnapshot) {
+    console.error('Failed to create linked doc from note');
+    return;
+  }
+  const linkedDoc = _doc.getStore({ id: doc.id });
   linkedDoc.load(() => {
     const rootId = linkedDoc.addBlock('affine:page', {
       title: new Text(docTitle),
     });
     linkedDoc.addBlock('affine:surface', {}, rootId);
-    const blockProps = getBlockProps(note);
-    // keep note props & show in both mode
-    const noteId = linkedDoc.addBlock(
-      'affine:note',
-      {
-        ...blockProps,
-        hidden: false,
-        displayMode: NoteDisplayMode.DocAndEdgeless,
-      },
-      rootId
-    );
-    // Add note to linked doc recursively
-    note.children.forEach(model => {
-      addBlocksToDoc(linkedDoc, model, noteId);
-    });
```

Comment:
This piece of code is also used in createLinkedDocFromEdgelessElements, which may lead to the same problem.

Reply:
OK